src/services/photos/network/PhotosNetworkManager.ts

```diff
@@ -31,33 +36,60 @@ export class PhotosNetworkManager implements RunnableService<PhotosNetworkManage
   private logger = photosLogger;
   // eslint-disable-next-line
   private onStatusChangeCallback: OnStatusChangeCallback = () => {};
-  private sdk: SdkManager;
+  // eslint-disable-next-line
+  private onUploadStartCallback: OnUploadStartCallback = () => {};
+  // eslint-disable-next-line
+  private onUploadProgressCallback: OnUploadProgressCallback = () => {};
   private options: { enableLog: boolean };
-  private queue = async.queue<PhotosNetworkOperation, Photo | null, Error>((task, next) => {
-    if (this.isAborted) {
-      throw new AbortedOperationError();
-    }
+  private queue = this.createQueue();
 
-    this.wrapWithDuration(task)
-      .then((result) => {
-        next(null, result);
-      })
-      .catch((err) => {
-        next(err, null);
-      });
-  }, PHOTOS_NETWORK_MANAGER_QUEUE_CONCURRENCY);
-
-  constructor(sdk: SdkManager, options = { enableLog: ENABLE_PHOTOS_NETWORK_MANAGER_LOGS }) {
-    this.sdk = sdk;
+  constructor(options = { enableLog: ENABLE_PHOTOS_NETWORK_MANAGER_LOGS }) {
     this.options = options;
-    this.queue.drain(() => {
+  }
+
+  private createQueue() {
+    const queue = async.queue<PhotosNetworkOperation, Photo | null, Error>((task, next) => {
+      if (this.isAborted) {
+        throw new AbortedOperationError();
```

Comment:
Is better to follow the same flow that the queue establishes, with is according to the async [docs](https://caolan.github.io/async/v3/docs.html#QueueObject) using the next callback with an error, in this case:

```
// maybe adding queue.kill() too could be a good idea.
return next(new AbortedOperationError());
```